### ISSUE TITLE
Playground

### ISSUE THREAD

**Issue by cojoj** [czechboy0/XcodeServerSDK#26]
I've investigated some projects which follows the pattern of having `.xcworkspace` in which they have **Framework** itself, sample project and also a Playground (using framework) and after digging I can confirm that we have everything to run this pattern!

Here's a proof:
![screen shot 2015-06-20 at 00 05 00](https://cloud.githubusercontent.com/assets/4522407/8263934/6715b5cc-16e0-11e5-9fad-c3fedfc0e5f2.png)
As you can see I can `import XcodeServerSDK` and use it in Playground with code completion etc.

Earlier the problem was with wrong configuration of Playground (was targeting iOS while framework support OS X only). I think we have solution to this _issue_ in #24 and also this matter was explained in by @esttorhe in #25.

I'll check this **PR** tomorrow and if everything will work fine I will strongly suggest to merge it.

**Comment by czechboy0** [comment]
Was this an iOS playground or an OS X one? I got this working on the OS X one only so far. See commit facb6cdcb6ecaddf5bda99b67c882f94aed2af24

**Comment by cojoj** [comment]
Nope, this was OS X playground... But I had the same error message:
![screen shot 2015-06-21 at 20 32 34](https://cloud.githubusercontent.com/assets/4522407/8272857/aed89d00-1854-11e5-8ed2-c9ac015001f8.png)
 From my POV, I also think that the issue is what @czechboy0 pointed in [this commit](https://github.com/czechboy0/XcodeServerSDK/commit/facb6cdcb6ecaddf5bda99b67c882f94aed2af24). On the other hand, do we really need to name `PRODUCT_NAME` differently? Can't we only use OS X or iOS playground (matter of choice) and have in mind that this is the issue? Final user won't have this problem as he/she will get the proper framework file.

I've also checked other well known frameworks to see how they handle this issue. For example [Alamofire](https://github.com/Alamofire/Alamofire) has internal Playground working only on iOS.

**Comment by esttorhe** [comment]
Personally I think both frameworks should be named the same to avoid conditional imports on a library using this as a dependency. 
But that's my point of view 

## 

Esteban Torres
(+506)8813-0934
Twitter: @esttorhe

On Sun, Jun 21, 2015 at 11:36 AM -0700, "Mateusz Zając" notifications@github.com wrote:

Nope, this was OS X playground... But I had the same error message:

 From my POV, I also think that the issue is what @czechboy0 pointed in this commit. On the other hand, do we really need to name PRODUCT_NAME differently? Can't we only use OS X or iOS playground (matter of choice) and have in mind that this is the issue? Final user won't have this problem as he/she will get the proper framework file.

**Comment by cojoj** [comment]
I agree with @esttorhe. What we want is to provide an easy way of importing `XcodeServerSDK` 😉

**Comment by czechboy0** [comment]
Sure, it's just annoying we'll have a playground that doesn't actually include our framework correctly (the iOS one). Let's leave it this way for now though and if anyone finds out a way to make both iOS and OS X playgrounds work, please comment on this issue. 

@cojoj Please close this if there's no other action required.

**Comment by esttorhe** [comment]
I'll play today a little bit and try to make it work as expected

**Comment by czechboy0** [comment]
:+1:

**Comment by esttorhe** [comment]
I made it work with some hacks here and there… :unamused: 

---

Here's the `iOS Playground` execution:
![screen shot 2015-06-22 at 18 20 08](https://cloud.githubusercontent.com/assets/2642850/8296252/abdfceea-190b-11e5-8ddc-6bc830e8b4ce.png)

# 

Here's the `OSX Playground` execution:
![screen shot 2015-06-22 at 18 21 33](https://cloud.githubusercontent.com/assets/2642850/8296254/b62b9e2e-190b-11e5-9a20-a82a332b2e4f.png)

---

I had to configure a custom build path on the workspace settings (I don't think this is needed but it was part of my explorations); biggest issues are that in order for the Playgrounds to work the deployment targets need to be downgraded:
- iOS: 7.0
- OSX: 10.9

Look at this: https://github.com/neonichu/quick-cli-runner/pull/3

Long story short; I don't think the benefit is worth the hassle; I'm going to push this to another branch in case you want to take a look at this implementation ¯\_(ツ)_/¯

**Comment by esttorhe** [comment]
Here's a fork with the workaround:
https://github.com/esttorhe/XcodeServerSDK/tree/playgrounds-workaround

In case you need to change the paths go to:

```
File → Workspace Settings → Advanced… → Custom → Relative to Workspace
```

```
• Products: Build/Products/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
• Intermediates: Build/Intermediates/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
```

But I think the complexity far outweighs the benefits IMHO

**Comment by cojoj** [comment]
@esttorhe thank for sorting it out... I think that if we have a _solution_ and we're aware of consequences we can agree that one Playground platform is sufficient for our usage 😉

**Comment by czechboy0** [comment]
Sounds good, let's remove the iOS playground from the project then? So that we don't have a failing playground there all the time. (a small PR with that removal and a declaration in our README should do)

Thanks guys for the effort, @esttorhe and @cojoj!

**Comment by esttorhe** [comment]
I agree; its better to nuke :fire: the `iOS` Playground and be done with this issue for now; if at some point there's a solution around playgrounds i'll update my branch
